### ISSUE TITLE
Cinnamon "burps" every about 30 seconds (part 2)

### ISSUE THREAD

**Issue by SjoerdV** [linuxmint/cinnamon#12095]
### Distribution

LMDE 6

### Package version

6.0.4

### Graphics hardware in use

Intel UHD 630 + NVIDIA GTX 1650 Mobile / Max-Q

### Frequency

Only occasionally

### Bug description

a short hang (including input devices and screens) every 30 seconds. No CPU spike or anything, just hangs for a short while. It goes away if I logout and login again. First I thought it was something intel/nvidia did with their modules/drivers, but it seems cinnamon does something unique here.

### Steps to reproduce

Very difficult! It is also intermittent: when I fiddle around with xrandr and when coming out of S3 sleep mode (maybe both are needed to trigger this issue, not sure). Best course of action is probably (my opinion, but I am no dev) to dig into the cinnamon code where it does something to hardware interrupts every 30 seconds.

I had this issue intermittently over the past few years myself since I started using Mint products but this existing issue (https://github.com/linuxmint/Cinnamon/issues/5532) from 7.5 years ago is from someone who had exactly the same issue all the way back then, so it does seem like a deep core issue.

### Expected behavior

no burps, hiccups, stalls, etc.

### Additional information

No journal logs show anything about the burp.

I only have 1 third party spice installed: 'Scripts Menu'
and 2 extensions: 'Blur Overview' and 'Transparent Panels'
no desklets

Restart Cinnamon has no effect. Logging in and out does have an effect.

Maybe remove the part when adding a new bug to Github "Please try searching for existing reports that may match the behavior you're seeing. If you find one, add to that issue instead of opening a new one. Include any relevant details that may differ from the original author's."
I tried to do exactly that (https://github.com/linuxmint/Cinnamon/issues/5532), but they said to open a new issue and locked the old one... not very good PR IMHO

**Comment by claudiux** [comment]
Maybe too many files in your Trash, and/or in ~/.cache subfolders. I cleaned them and Cinnamon is smoother.

**Comment by DirkHaar** [comment]
I had this behavior with a former notebook due to a damaged USB port, it also can be generated by the so-called (keyboard) flexing problem on some Thinkpad models. Processor is halted the moment you attach a USB device, if damaged (cold solder joint problem) it may take longer. This happened on  or XP (not use if also on Mint (17 or 18) then, but this shouldn't matter as it is a hardware interrupt.
In my case, I type up to three, four words before recognizing, the next one appeared as normal, quite annoying (though better than crashes).
You can check this by not using USB devices and simply let your computer play a video.
Use your keyboard then in an editor for further testing.

**Comment by SjoerdV** [comment]
> I had this behavior with a former notebook due to a damaged USB port, it also can be generated by the so-called (keyboard) flexing problem on some Thinkpad models. Processor is halted the moment you attach a USB device, if damaged (cold solder joint problem) it may take longer. This happened on  or XP (not use if also on Mint (17 or 18) then, but this shouldn't matter as it is a hardware interrupt.
> 
> In my case, I type up to three, four words before recognizing, the next one appeared as normal, quite annoying (though better than crashes).
> 
> You can check this by not using USB devices and simply let your computer play a video.
> 
> Use your keyboard then in an editor for further testing.

@DirkHaar, Thanks for your suggestion

I doubt it is a hardware failure in my case because of the exact 30 seconds intervals AND the fact this does not occur (ever!) when using Windows and XFCE (also using Xorg, mutter and lightdm). 

There is a reason I created this issue in the cinnamon project, because I already eliminated most other possibilities 😉

**Comment by DirkHaar** [comment]
I missed exact 30sec if you wrote that...
Something WiFi related? A cronjob you forgot?

**Comment by SjoerdV** [comment]
> I missed exact 30sec if you wrote that...
> Something WiFi related? A cronjob you forgot?

@DirkHaar Nope, I run wired only (wifi is disabled). Also no cronjobs that could possibly trigger a full system interrupt. And I don't have any such job running every 30 seconds anyway (both user and root ones)

**Comment by claudiux** [comment]
Which version of linux?
`uname -r`

**Comment by SjoerdV** [comment]
> Which version of linux?
> 
> `uname -r`

@claudiux Kernel: 6.1.0-18-amd64

Maybe the best way to close this issue is to have someone with intricate knowledge of the cinnamon codebase confirm there are no 'timed/triggered subprocesses that have system interrupt capabilities'. That would satisfy me I have to find the culprit of mentioned symptoms elsewhere.

Thanks again!

**Comment by just-max** [comment]
Is this is on a laptop and does it only happen with the lid closed? Then it sounds like https://github.com/linuxmint/cinnamon/issues/7184, which is triggered exactly every 30 seconds

**Comment by SjoerdV** [comment]
Yes and yes, so I can totally agree to close this issue in favor of the one you mentioned.
